generator/src/main/java/com/scottlogic/deg/generator/generation/databags/StandardRowSpecDataBagSourceFactory.java

```diff
@@ -13,37 +13,26 @@
 
 public class StandardRowSpecDataBagSourceFactory implements RowSpecDataBagSourceFactory {
     private final FieldSpecValueGenerator generator;
+    private final GenerationConfig config;
 
     @Inject
-    public StandardRowSpecDataBagSourceFactory (
-        FieldSpecValueGenerator generator) {
+    public StandardRowSpecDataBagSourceFactory(
+        FieldSpecValueGenerator generator,
+        GenerationConfig config) {
```

Comment:
I'm not a fan of this. Basic utility classes shouldn't be peeking at top-level config, they should just be told explicitly how to behave. E.g. in this case I think the most elegant option is a parameter on `createDataBags`.

Reply:
as in, you're not a fan of config being passed around?

it could be changed to just pass in the combination strategy.

Reply:
Yeah, it should consume just a combination strategy - the next question is whether a constructor or method parameter (my preference is the latter, 'cos combination strategy is a functional choice and dependencies should be opinionated about the _how_, not the _what_).

Reply:
Injected now.

Reply:
> the next question is whether a constructor or method parameter (my preference is the latter, 'cos combination strategy is a functional choice and dependencies should be opinionated about the how, not the what)

:stuck_out_tongue:

Reply:
Im not changing how we do IOC in this pr. Thats a much larger discussion